relational_operators/DeleteOperator.cpp

```diff
@@ -81,19 +85,25 @@ void DeleteWorkOrder::execute(QueryContext *query_context,
 
   // TODO(harshad): Stream the block ID only if the predicate returned at least
   // one match in the StorageBlock.
-  ForemanMessage message(ForemanMessage::DataPipelineMessage(delete_operator_index_,
-                                                             input_block_id_,
-                                                             rel_id_));
+  serialization::DataPipelineMessage proto;
+  proto.set_operator_index(delete_operator_index_);
+  proto.set_block_id(input_block_id_);
+  proto.set_relation_id(rel_id_);
+
+  const std::size_t proto_length = proto.ByteSize();
+  char *proto_bytes = static_cast<char*>(std::malloc(proto_length));
```

Comment:
@pateljm Well, I should have pointed out that the Google protobuf example used to have the following code, until using `SerializeToOstream` instead. But `SerializeToArray` fits our case, as `SerializeToOstream` introduces an extra overhead for converting `std::string` to a c-like string.

```
int size = address_book.ByteSize(); 
void *buffer = malloc(size);
address_book.SerializeToArray(buffer, size);
```

Using both the heap and stack does not mean we are not **consistent** for memory allocation, because we use the heap for a variable-length, out-of-band data (there is no safe and efficient way to allocate them on the stack in c++), while the stack for the rest.

And yes, I release the heap memory immediately after use in the same function.